File: panaderia-backend/src/main/java/com/besoft/panaderia/service/VentaService.java
package com.besoft.panaderia.service;

import java.util.List;

import com.besoft.panaderia.dto.request.VentaBuscarRequest;
import com.besoft.panaderia.dto.request.VentaRequest;
import com.besoft.panaderia.dto.response.OutResponse;
import com.besoft.panaderia.dto.response.VentaResponse;

public interface VentaService {

	public OutResponse<VentaResponse> registrarVenta(VentaRequest c);
	
	public OutResponse<?> imprimirTicketVenta(VentaRequest c);

	public OutResponse<List<VentaResponse>> listarVenta(VentaBuscarRequest req);
}

File: panaderia-frontend/src/main/panaderia-frontend/src/app/modules/intranet/services/venta.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { VentaBuscarRequest } from '../dto/request/venta-buscar.request';
import { VentaRequest } from '../dto/request/venta.request';
import { OutResponse } from '../dto/response/out.response';
import { VentaResponse } from '../dto/response/venta.response';

@Injectable()
export class VentaService {

  constructor(private http: HttpClient) { }

  public listarVenta(req: VentaBuscarRequest): Observable<OutResponse<VentaResponse[]>> {
    return this.http.post<OutResponse<VentaResponse[]>>(`${environment.WsPanaderiaBackend}/venta/listarVenta`, req);
  }

  public registrarVenta(req: VentaRequest): Observable<OutResponse<VentaResponse>> {
    return this.http.post<OutResponse<VentaResponse>>(`${environment.WsPanaderiaBackend}/venta/registrarVenta`, req);
  }

  public imprimirTicketVenta(req: VentaRequest): Observable<OutResponse<VentaResponse>> {
    return this.http.post<OutResponse<VentaResponse>>(`${environment.WsPanaderiaBackend}/venta/imprimirTicketVenta`, req);
  }
}
